lib/msf/base/sessions/command_shell.rb

```diff
@@ -487,7 +487,7 @@ def run_single(cmd)
     end
 
     # User input is not a built-in command, write to socket directly
-    shell_write(cmd + "\n")
+    shell_write(cmd.chomp + "\n")
```

Comment:
Hmm, since this is fixing up newlines on user input I think I would prefer adding the `#chomp` on https://github.com/rapid7/metasploit-framework/blob/c6f1c107375c4b6b3d5fbdc6e8d5af2bbbf6c811/lib/msf/base/sessions/command_shell.rb#L655

Also, `#chomp` will remove `\r`s if they're present at the end of the line. Does that mess up Windows installs or shells?

Reply:
I wanted to normalize all inputs, since `run_single` methods typically have multiple users, but since this implementation is unique, I think it's fine to address the specific inconsistency.

Windows shells should be fine, since we've always sent just a newline. Windows installs should be okay, but I'm unable to test that.

Reply:
tl;dr I'd rather have bugs pop up than lose sight of them through this normalization.